src/main/java/rx/internal/operators/OperatorOnBackpressureBuffer.java

```diff
@@ -27,15 +23,93 @@
 import rx.functions.Action0;
 import rx.internal.util.BackpressureDrainManager;
 
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
 public class OperatorOnBackpressureBuffer<T> implements Operator<T, T> {
 
     private final Long capacity;
     private final Action0 onOverflow;
+    private final BackpressureOverflowStrategy overflowStrategy;
 
     private static class Holder {
         static final OperatorOnBackpressureBuffer<?> INSTANCE = new OperatorOnBackpressureBuffer<Object>();
     }
-    
+
+    /**
+     * A convenience class to provide singleton instances of available {@code BackpressureOverflowStrategy}.
+     */
+    @SuppressWarnings("unused")
+    public static class OverflowStrategy {
```

Comment:
Once `BackpressureOverflowStrategy` has been moved to `rx` package, the values of this class can become its constants, but all of them should have the type `BackpressureOverflowStrategy`. The type after the assignment can stay the same as now and can point to the current internal location.